src/main/scala/sigmastate/lang/Terms.scala

```diff
@@ -31,7 +33,7 @@ object Terms {
     def tpe: SType = givenType ?: body.tpe
   }
   object Let {
-    def apply(name: String, value: SValue): Let = Let(name, NoType, value)
+    def apply(name: String, value: SValue): Let = mkLet(name, NoType, value).asInstanceOf[Let]
```

Comment:
@greenhat Shouldn't use builder.

Reply:
@ergomorphic Good catch! Sorry, missed this one. Fixed.